hcl2template/types.packer_config.go

```diff
@@ -311,13 +352,92 @@ func (cfg *PackerConfig) evaluateDatasources(skipExecution bool) hcl.Diagnostics
 			})
 			continue
 		}
+
 		ds.value = realValue
 		cfg.Datasources[ref] = ds
 	}
 
+	// Now that most of our data sources have been started and executed, we can
+	// try to execute the ones that depend on other data sources.
+	for ref := range dependencies {
```

Comment:
This could be executing stuff that we want to be executed once, more than once, I think that could cause issues with our users down the line.

One thing we could do here would be to add a function to a data source that says whether 'all parents have been evaluated'.